From b0e7038adb00ea25c3dd7abe6a4799e99f6d944e Mon Sep 17 00:00:00 2001
From: Daniel Goehring <dgoehrin@os.amperecomputing.com>
Date: Fri, 12 Mar 2021 17:46:20 -0500
Subject: [PATCH 15/32] QS|MQ: Boundary-Scan user config files [DO NOT
 UPSTREAM]

Add boundary-scan user configuration files for Ampere Altra ("Quicksilver")
and Ampere Altra Max ("Mystique")

Configured for the Olimex ARM-USB-OCD-H JTAG probe.
https://www.olimex.com/Products/ARM/JTAG/ARM-USB-OCD-H/

OpenOCD configured for the following Ampere Altra and Ampere Altra Max options
1. One and two socket platforms

Change-Id: I5ac67718692293f722d1c787820f89bc61888503
Signed-off-by: Daniel Goehring <dgoehrin@os.amperecomputing.com>
---
 tcl/openocd.mq_1s_bs.cfg | 60 ++++++++++++++++++++++++++++++++++++++++
 tcl/openocd.mq_2s_bs.cfg | 60 ++++++++++++++++++++++++++++++++++++++++
 tcl/openocd.qs_1s_bs.cfg | 59 +++++++++++++++++++++++++++++++++++++++
 tcl/openocd.qs_2s_bs.cfg | 59 +++++++++++++++++++++++++++++++++++++++
 4 files changed, 238 insertions(+)
 create mode 100644 tcl/openocd.mq_1s_bs.cfg
 create mode 100644 tcl/openocd.mq_2s_bs.cfg
 create mode 100644 tcl/openocd.qs_1s_bs.cfg
 create mode 100644 tcl/openocd.qs_2s_bs.cfg

diff --git a/tcl/openocd.mq_1s_bs.cfg b/tcl/openocd.mq_1s_bs.cfg
new file mode 100644
index 000000000..ecce3e618
--- /dev/null
+++ b/tcl/openocd.mq_1s_bs.cfg
@@ -0,0 +1,60 @@
+#
+# Example OpenOCD user configuration for the Ampere Altra Max ("Mystique")
+# Development Platform
+#
+# Copyright (c) 2020-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "5000" kHz
+# Syntax: -c "set JTAGFREQ {5000}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "mq"
+# Syntax: -c "set SYSNAME {mq}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/ftdi/olimex-arm-usb-ocd-h.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 5000
+}
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME mq
+}
+
+set MQ_ENABLE ""
+source [find board/ampere_qs_mq_1s_bs.cfg]
diff --git a/tcl/openocd.mq_2s_bs.cfg b/tcl/openocd.mq_2s_bs.cfg
new file mode 100644
index 000000000..b59156a55
--- /dev/null
+++ b/tcl/openocd.mq_2s_bs.cfg
@@ -0,0 +1,60 @@
+#
+# Example OpenOCD user configuration for the Ampere Altra Max ("Mystique")
+# Development Platform
+#
+# Copyright (c) 2020-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "5000" kHz
+# Syntax: -c "set JTAGFREQ {5000}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "mq"
+# Syntax: -c "set SYSNAME {mq}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/ftdi/olimex-arm-usb-ocd-h.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 5000
+}
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME mq
+}
+
+set MQ_ENABLE ""
+source [find board/ampere_qs_mq_2s_bs.cfg]
diff --git a/tcl/openocd.qs_1s_bs.cfg b/tcl/openocd.qs_1s_bs.cfg
new file mode 100644
index 000000000..5b915e77c
--- /dev/null
+++ b/tcl/openocd.qs_1s_bs.cfg
@@ -0,0 +1,59 @@
+#
+# Example OpenOCD user configuration for the Ampere Altra ("Quicksilver")
+# Development Platform
+#
+# Copyright (c) 2019-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "5000" kHz
+# Syntax: -c "set JTAGFREQ {5000}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "qs"
+# Syntax: -c "set SYSNAME {qs}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/ftdi/olimex-arm-usb-ocd-h.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 5000
+}
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME qs
+}
+
+source [find board/ampere_qs_mq_1s_bs.cfg]
diff --git a/tcl/openocd.qs_2s_bs.cfg b/tcl/openocd.qs_2s_bs.cfg
new file mode 100644
index 000000000..aae5bd4ac
--- /dev/null
+++ b/tcl/openocd.qs_2s_bs.cfg
@@ -0,0 +1,59 @@
+#
+# Example OpenOCD user configuration for the Ampere Altra ("Quicksilver")
+# Development Platform
+#
+# Copyright (c) 2019-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "5000" kHz
+# Syntax: -c "set JTAGFREQ {5000}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "qs"
+# Syntax: -c "set SYSNAME {qs}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/ftdi/olimex-arm-usb-ocd-h.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 5000
+}
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME qs
+}
+
+source [find board/ampere_qs_mq_2s_bs.cfg]
-- 
2.25.1

